tests/functional/test_download.py

```diff
@@ -145,6 +145,19 @@ def test_download_temporary_file_does_not_exist(self):
         possible_matches = glob.glob('%s*' % self.filename + os.extsep)
         self.assertEqual(possible_matches, [])
 
+    def test_download_for_fileobj(self):
```

Comment:
Could you add a test for a non-file file-like object, such as `StringIO` or `BytesIO`?